src/WebJobs.Script/Workers/SharedMemoryDataTransfer/SharedMemoryManager.cs

```diff
@@ -0,0 +1,314 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO.MemoryMappedFiles;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.SharedMemoryDataTransfer
+{
+    /// <summary>
+    /// Manages accesses to <see cref="SharedMemoryMap"/>.
+    /// Also controls lifetime of <see cref="SharedMemoryMap"/> allocated from this component by adding/removing references.
+    /// </summary>
+    public class SharedMemoryManager : ISharedMemoryManager
+    {
+        private readonly ILoggerFactory _loggerFactory;
+
+        private readonly ILogger _logger;
+
+        private readonly IMemoryMappedFileAccessor _mapAccessor;
+
+        public SharedMemoryManager(ILoggerFactory loggerFactory, IMemoryMappedFileAccessor mapAccessor)
+        {
+            _loggerFactory = loggerFactory;
+            _logger = _loggerFactory.CreateLogger<SharedMemoryManager>();
+            _mapAccessor = mapAccessor;
+            AllocatedSharedMemoryMaps = new ConcurrentDictionary<string, SharedMemoryMap>();
+            InvocationSharedMemoryMaps = new ConcurrentDictionary<string, HashSet<string>>();
+        }
+
+        /// <summary>
+        /// Gets mapping of invocation IDs to list of names of shared memory maps allocated for that invocation.
+        /// </summary>
+        public ConcurrentDictionary<string, HashSet<string>> InvocationSharedMemoryMaps { get; private set; }
```

Comment:
Seems like these dictionaries should be private.

Reply:
Kept them as public properties for unit testing. Any better pattern I can use for that?

Reply:
Make them internal?